File: docs/js_vs_py/02_closures.py
class ClassWithProperties:
    privateValue = 'Can not be changed'
    def expose(self):
        print(self.privateValue)


def closure():
    privateValue = 'In a closure, it can not be changed'
    def expose():
        print privateValue

    return expose  ## Returns a function


instance = ClassWithProperties()
instance.expose()
instance.privateValue = 'But it was changed'
instance.expose()


closureResult = closure()
closureResult()
closureResult.privateValue = 'This change will not work'
closureResult()

File: docs/js_vs_py/02_closures.js
"use strict";

class MyClass {
    constructor(){
        this.myAttribute = 'Can not be changed';
    }
    
    expose() {
        console.log(this.myAttribute)
    }
    
}


const instance = new MyClass();
instance.expose()
instance.myAttribute = 'But it was changed'
instance.expose()


const closure = () => {
    let myPrivateVariable = 'It can not be changed'

    const expose = () => {
        console.log(myPrivateVariable);
    }

    return expose;
}

const closureResult = closure()
closureResult()
closureResult.myPrivateVariable = 'But I changed it'
closureResult()

